enhancements/operator-conditions-lib.md

```diff
@@ -0,0 +1,116 @@
+---
+title: operator-conditions-lib
+authors:
+  - "@varshaprasad96"
+reviewers:
+  - "@joelanford"
+  - "@estroz"
+  - "@awgreene"
+approvers:
+  - "@joelanford"
+  - "@estroz"
+  - "@awgreene"
+creation-date: 2020-09-28
+last-updated: 2020-09-29
+status: implementable
+---
+
+# Operator Conditions API Library
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+The goal of this enhancement is to define the deliverables of a library which would provide APIs for operators to access the [Condition CRD][conditions_proposal] created by OLM. Using this library, operators can specify or update conditions which could influence the behavior of OLM.
+
+## Motivation
+
+[Operator Lifecycle Manager][olm] intends to provide operators with a channel to communicate complex states that influence its behavior while managing the operator. To do so, OLM will provide a [`Condition`][conditions_proposal] Custom Resource Definition (CRD) for the operator. Based on the [conditions][conditions] set in the CR, OLM will change its behavior accordingly.
+
+Operators need to be able to read the Condition CR created by OLM, and also have the ability to add/update necessary conditions. This library will provide APIs, using which users can read the Condition CRD present in the cluster for their operator and modify the `.status.conditions` array to add or update any specific condition.
+
+### Goals:
+
+- List the deliverables of the library.
+
+### Non-Goals:
+
+- Discuss on how OLM will process the conditions set in the CRD.
+- List the function signatures/interface types which will be present in the library.
+
+### User Stories
+
+#### Story 1
+
+As an operator author, I want my operator to be able to add/modify conditions in the `Condition CR`.
+
+#### Story 2
+
+As an operator author, I would like to have the flexibility of being able to read the `Condition CR` and decide to not set any conditions, thereby opting for default OLM behavior.
```

Comment:
```suggestion
As an operator author, I would like to have the flexibility of being able to read the `Condition` CR and decide to not set any conditions, thereby opting for default OLM behavior.
```

Reply:
Why would an operator want to get the `Condition` if it wasn't going to modify it?

Reply:
Have reworded this. What I meant was, operators should have the flexibility to view the entire CR